contracts/peripheral/GaugeVoteRewards.sol

```diff
@@ -0,0 +1,318 @@
+            uint256 pid = _poolIds[i];
+            IBooster.PoolInfo memory poolInfo = booster.poolInfo(pid);
+            getPoolId[poolInfo.gauge] = pid;
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       View 
+    ------------------------------------------------------------------- */
+
+    function getCurrentEpoch() external view returns (uint256) {
+        return _getCurrentEpoch();
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice  Wraps the booster.voteGaugeWeight call to track weights for each epoch
+     *          So AURA rewards can be distributed pro rata to those pool stashes
+     * @param _gauge    Array of the gauges
+     * @param _weight   Array of the weights
+     * @return bool for success
+     */
+    function voteGaugeWeight(address[] calldata _gauge, uint256[] calldata _weight) external onlyOwner returns (bool) {
+        uint256 totalWeight = 0;
+        uint256 totalDepositWeight = 0;
+        uint256 epoch = _getCurrentEpoch();
+
+        // Loop through each gauge and store it's weight for this epoch
+        for (uint256 i = 0; i < _gauge.length; i++) {
+            address gauge = _gauge[i];
+            uint256 weight = _weight[i];
+
+            totalWeight = totalWeight.add(weight);
+
+            // Some gauges like veBAL have no deposits so we just skip
+            // those special cases
+            if (isNoDepositGauge[gauge]) continue;
+
+            getWeightByEpoch[epoch][gauge] = weight;
+            totalDepositWeight = totalDepositWeight.add(weight);
+        }
+
+        // Update the total weight for this epoch
+        getTotalWeight[epoch] = getTotalWeight[epoch].add(totalWeight);
+        require(totalWeight == 10000, "!totalWeight");
+
+        // Forward the gauge vote to the booster
+        return booster.voteGaugeWeight(_gauge, _weight);
+    }
+
+    /**
+     * @notice Process gauge rewards for the given epoch for mainnet
+     * @param _gauge Array of gauges
+     * @param _epoch The epoch
+     */
+    function processGaugeRewards(address[] calldata _gauge, uint256 _epoch) external onlyDistributor {
+        require(_epoch <= _getCurrentEpoch(), "!epoch");
+
+        for (uint256 i = 0; i < _gauge.length; i++) {
+            address gauge = _gauge[i];
+            uint16 dstChainId = getDstChainId[gauge];
+
+            // This is not a current chain gauge and should be processed by
+            // processSidechainGaugeRewards instead this also covers cases
+            // where a src chain has not been set for invalid gauges
+            require(dstChainId == lzChainId, "dstChainId!=lzChainId");
+
+            uint256 amountToSend = _getAmountToSend(_epoch, gauge);
+
+            // Fund the extra reward distro for the next 2 epochs
+            uint256 pid = getPoolId[gauge];
```

Comment:
if a gauge has not been configured on ` uint256 pid = getPoolId[gauge];` it will process the reward to PID = 0 , it is prone to errors